packages/quantic/force-app/main/default/lwc/quanticDidYouMean/quanticDidYouMean.js

```diff
@@ -84,7 +84,11 @@ export default class QuanticDidYouMean extends LightningElement {
    */
   initialize = (engine) => {
     this.headless = getHeadlessBundle(this.engineId);
-    this.didYouMean = this.headless.buildDidYouMean(engine);
+    this.didYouMean = this.headless.buildDidYouMean(engine, {
+      options: {
+        queryCorrectionMode: 'legacy',
+      },
+    });
```

Comment:
@coveo/salesforce-integration  hi can someone help me understand why the tests are failing ? Shouldn't this work the same as before ?

https://github.com/coveo/ui-kit/actions/runs/11240072591/job/31249748165?pr=4495

Reply:
The only change in headless is this : 

<img width="747" alt="image" src="https://github.com/user-attachments/assets/e6ae4d83-3afd-484e-85e5-890bcc974865">

Reply:
in the Insight Headless Did You mean controller, there was a bug where no options were passed to the core controller so  the `queryCorrectionMode ` you setting was no applied in the insight use case.

The following changes in the `headless-insight-did-you-mean.ts` should do it:
<img width="1728" alt="Screenshot 2024-10-08 at 3 19 43 PM" src="https://github.com/user-attachments/assets/4519c079-50a2-49c9-830e-333573f3e7ce">

Reply:
I applied the fixes but do you think it would be better to pass `legacy` directly to the core controller in the insight one ?

  `const controller = buildCoreDidYouMean(engine, {options: {queryCorrectionMode: 'legacy'});
`

Reply:
indeed yes as the `next` method is not even yet supported in the insight use case 👍